ConsoleSession.java

```diff
@@ -97,10 +99,16 @@
         this.consoleReader.setPrompt(ANSI_PURPLE + session.keyspace().name() + ANSI_RESET + "> ");
         this.printErr = printErr;
 
-        File file = new File(HISTORY_FILE);
-        file.createNewFile();
-        this.historyFile = new FileHistory(file);
-        this.consoleReader.setHistory(this.historyFile);
+        PersistentHistory history;
+        try {
+            File file = new File(HISTORY_FILE);
```

Comment:
Here you can maybe use `File.exists()` rather then creating a `history = new FlushableMemoryHistory();` in the `catch` block

Reply:
Or you can also keep the catch, but I think it's important that we log with a warning the fact that we are falling back to an in-memory history, so maybe just add in the catch:
```
LOG.warn("An in-memory history will be used due to exception raised while trying to access history file: ", e.getMessage());
history = new FlushableMemoryHistory();
```

Reply:
@marco-scoppetta I think we actually want to fallback on _any_ case on not being able to create/open a file. How could I utilize `File.exists` if creating a new file triggers `IOException`?

Reply:
@vmax https://github.com/graknlabs/console/pull/16#discussion_r344146485